### PR TITLE
Add locked deck feature for events

### PR DESCRIPTION
**Pull request by EvadLhorg**

### CODE REVIEW

File: Components/Decks/DeckEditor.jsx [hunk 1]
```diff
@@ -388,6 +395,12 @@ class DeckEditor extends React.Component {
                     </Typeahead>
                     <TextArea label='Cards' labelClass='col-sm-3' fieldClass='col-sm-9' rows='10' value={ this.state.cardList }
                         onChange={ this.onCardListChange.bind(this) } />
+                    { this.state.eventId && 
+                        <h4>Please be aware: When you select this event and save this deck, you are locking it from further changes and you will not be able to edit this deck anymore</h4>
```
Comment (cryogen) [inline Components/Decks/DeckEditor.jsx:399]: "Assigning this deck to an event will mean that you will be unable to modify or delete this deck for the duration of the event"

The last part here isn't currently true, but probably should be

File: Components/Decks/DeckRow.jsx [hunk 2]
```diff
@@ -16,6 +16,14 @@ class DeckRow extends React.Component {
         }
     }
 
+    displayNameForDeck(deck) {
+        if(deck.eventId) {
+            let deckEvent = this.props.events.find(e => e._id === deck.eventId);
+            return '(' + deckEvent.name + ') ' + deck.name;
```
Comment (cryogen) [inline Components/Decks/DeckRow.jsx:22]: Use string interpolation please (``)

File: pages/EventsAdmin/EventEditor.jsx [hunk 3]
```diff
@@ -269,6 +272,9 @@ class EventEditor extends React.Component {
                         value={ this.state.format }
                         onChange={ this.onChange.bind(this, 'format') } />
 
+                    <Checkbox name='lockDecks' label='Lock Decks' labelClass='col-sm-4' fieldClass='col-sm-offset-3 col-sm-8'
```
Comment (cryogen) [inline pages/EventsAdmin/EventEditor.jsx:275]: "Prevent users from making changes to their decks for the duration of the event"

File: Components/Decks/DeckEditor.jsx (hunk 1 above)
Comment (EvadLhorg) [reply Components/Decks/DeckEditor.jsx:399]: good idea, changed it

File: Components/Decks/DeckRow.jsx (hunk 2 above)
Comment (EvadLhorg) [reply Components/Decks/DeckRow.jsx:22]: fixed

File: pages/EventsAdmin/EventEditor.jsx (hunk 3 above)
Comment (EvadLhorg) [reply pages/EventsAdmin/EventEditor.jsx:275]: changed it

**Review by cryogen — APPROVED** [review]

**Pull request merged**